server/src/main/java/org/opensearch/index/store/Store.java

```diff
@@ -174,6 +174,7 @@ public class Store extends AbstractIndexShardComponent implements Closeable, Ref
 
     private final AtomicBoolean isClosed = new AtomicBoolean(false);
     private final StoreDirectory directory;
+    private final StoreDirectory secondaryDirectory;
```

Comment:
can story not have to worry about how many directories are there and just have one StoreDirectory which is extended by a Remote+Local store Directory?

Reply:
https://github.com/opensearch-project/OpenSearch/pull/3285#discussion_r870314561

Reply:
Please check the latest changes. I have removed reference to secondaryDirectory from the `Store`.